tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -6449,6 +6456,11 @@ bool HLSLExternalSource::MatchArguments(
              "have as many arguments and types as the intrinsic template");
   }
 
+  // For object return types that need to match arguments, we need to slot in the full type here
+  // Can't do it sooner because when return is encountered above, the other arg types haven't been set
```

Comment:
nit: can you clang format your changes?

Reply:
I did it. It changed more than I expected. It's all okay except for the way they shifted the AnySampler and ObjectTT initialization, which I would argue is harder to read and inconsistent with all the others, so I reverted that part.